### PR TITLE
prepare a FEDRAMP env variable for use

### PR DESCRIPTION
**Pull request by yithian**
for [OSD-0249](https://issues.redhat.com/browse/OSD-0249)

### CODE REVIEW

**Comment by yithian** [comment]
the lint failure is because the fedramp variable isn't being used yet

File: hack/olm-registry/olm-artifacts-template.yaml [hunk 1]
```diff
@@ -10,6 +10,8 @@ parameters:
   required: true
 - name: CHANNEL
   value: staging
+- name: FEDRAMP
+  required: true
```
Comment (maorfr) [inline hack/olm-registry/olm-artifacts-template.yaml:14]: i recommend not making this required, and set the default value to be `"false"`.

File: hack/test/deploy/deploy.yaml [hunk 2]
```diff
@@ -32,3 +32,5 @@ spec:
                   fieldPath: metadata.name
             - name: OPERATOR_NAME
               value: "certman-operator"
+            - name: FEDRAMP
+              value: false
```
Comment (maorfr) [inline hack/test/deploy/deploy.yaml:36]: value should probably be `"false"`

File: hack/test/deploy/deploy.yaml [hunk 3]
```diff
@@ -32,3 +32,5 @@ spec:
                   fieldPath: metadata.name
             - name: OPERATOR_NAME
               value: "certman-operator"
+            - name: FEDRAMP
+              value: "false"
```
Comment (maorfr) [inline hack/test/deploy/deploy.yaml:36]: `value: "${FEDRAMP}"`?

File: pkg/controller/certificaterequest/certificaterequest_controller.go [hunk 4]
```diff
@@ -55,6 +56,7 @@ const (
 	certRequestSuffix                     = "-primary-cert-bundle"
 )
 
+var fedramp = os.Getenv("FEDRAMP") == "true"
```
Comment (dustman9000) [inline pkg/controller/certificaterequest/certificaterequest_controller.go:59]: Can we add a `log.Info` message for the presence of this var? Also, should "FEDRAMP" be a const?

Comment (yithian) [reply pkg/controller/certificaterequest/certificaterequest_controller.go:59]: i tried defining it as a const but that threw an error: `os.Getenv("FEDRAMP") == "true" (untyped bool value) is not constant`

File: pkg/controller/certificaterequest/certificaterequest_controller.go [hunk 5]
```diff
@@ -118,6 +120,11 @@ func (r *ReconcileCertificateRequest) Reconcile(request reconcile.Request) (reco
 	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
 
 	reqLogger.Info("reconciling CertificateRequest")
+	if len(os.Getenv("FEDRAMP")) > 0 {
```
Comment (dustman9000) [inline pkg/controller/certificaterequest/certificaterequest_controller.go:123]: Could this just be the `fedramp` var?

File: hack/olm-registry/olm-artifacts-template.yaml [hunk 6]
```diff
@@ -10,6 +10,8 @@ parameters:
   required: true
 - name: CHANNEL
   value: staging
+- name: FEDRAMP
+  value: "${FEDRAMP}"
```
Comment (maorfr) [inline hack/olm-registry/olm-artifacts-template.yaml:14]: This needs to be "false"

Comment (yithian) [reply hack/olm-registry/olm-artifacts-template.yaml:14]: https://github.com/openshift/certman-operator/pull/209 opened to address this

**Pull request merged**